src/containers/ConnectorManagement.jsx

```diff
@@ -171,19 +154,55 @@ export default class ConnectorManagement extends Component {
         }
       })
       .catch(error => { // eslint-disable-line
-        this.setState({ submitting: false })
-        console.error(error)
         Notifier.error(t('account config error'))
         this.gotoParent()
         throw error
       })
   }
 
-  connectAccountOAuth (values) {
-    return this._updateAccount(0, values)
-      .then(() => {
-        window.open(prepareConnectURL(this.state.connector), 'width=800,height=800')
+  terminateOAuth (storageEvent) {
+    const { t } = this.context
+    if (storageEvent.key !== 'oauth_terminating') return // ignore other keys
+    // get account id from localStorage event and remove the listener
+    const accountID = JSON.parse(storageEvent.newValue).key
+    window.removeEventListener('storage', this.terminateOAuth.bind(this))
+
+    // update connector to get the new account
+    this.setState({submitting: true})
+    this.store.fetchKonnectorInfos(this.props.params.connectorSlug)
+      .then(konnector => {
+        return this.store
+          .fetchAccounts(this.props.params.connectorSlug, null)
+          .then(accounts => {
+            konnector.accounts = accounts
+            const currentIdx = accounts.findIndex(a => a._id === accountID)
+            const folderPath = t('konnector default base folder', konnector)
+            return this.runConnection(
+              {oauth: accounts[currentIdx].oauth},
+              folderPath
+            ).then(() => {
+              this.setState({
+                connector: konnector,
+                isConnected: konnector.accounts.length !== 0,
+                selectedAccount: currentIdx,
+                submitting: false
+              })
+            }).catch(
+              error => this.setState({submitting: false, error: error.message})
+            )
+          })
       })
+      .catch(error => {
+        this.setState({submitting: false, error: error.message})
+      })
+  }
+
+  connectAccountOAuth (accountType) {
+    const cozyUrl =
+      `${window.location.protocol}//${document.querySelector('[role=application]').dataset.cozyDomain}`
+    const newTab = window.open('', 'popup', 'width=800,height=800')
+    newTab.location.href = `${cozyUrl}/accounts/${accountType}/start`
```

Comment:
Just to check. Did you see this with romain? The url for the popup is now related to the account type?